cardinal/ecs/world.go

```diff
@@ -469,8 +469,9 @@ func (w *World) Tick(ctx context.Context) error {
 		queue: txs,
 	}
 
-	for _, sys := range w.systems {
-		if err := sys(w, txQueue); err != nil {
+	for i, sys := range w.systems {
+		systemLogger := w.logger.CreateSystemLogger(w.systemNames[i])
```

Comment:
An advantage of the previous pattern was that this systemLogger was only being created once for each registered system. In this case we're creating new instances for each tick. Can a systemLogger slice replace the systemNames slice so we allocate them just once?

Reply:
good idea

Reply:
added